pkg/pf/tests/internal/cross-tests/tfwrite.go

```diff
@@ -1,173 +1,182 @@
 package crosstests
 
 import (
-	"fmt"
-	"io"
-	"sort"
-
-	"github.com/hashicorp/hcl/v2/hclwrite"
 	pschema "github.com/hashicorp/terraform-plugin-framework/provider/schema"
 	rschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	crosstestsimpl "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/internal/tests/cross-tests/impl"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
-	"github.com/zclconf/go-cty/cty"
 )
 
-// Provider writes a provider declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeProvider(out io.Writer, sch pschema.Schema, providerName string, config map[string]cty.Value) error {
-	if !cty.ObjectVal(config).IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
-	}
-	f := hclwrite.NewEmptyFile()
-	block := f.Body().AppendNewBlock("provider", []string{providerName})
-	writePfProvider(block.Body(), sch, config)
-	_, err := f.WriteTo(out)
-	return err
-}
-
-// Resource writes a resource declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeResource(
-	out io.Writer, sch rschema.Schema, resourceType, resourceName string, config map[string]cty.Value,
-) error {
-	if !cty.ObjectVal(config).IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
+type pfNestingMode uint8
+
+const (
+	pfNestingModeUnknown pfNestingMode = 0
+	pfNestingModeList    pfNestingMode = 1
+	pfNestingModeSet     pfNestingMode = 2
+	pfNestingModeSingle  pfNestingMode = 3
```

Comment:
Unless the numbers matter (which they don't), then lets use `iota`.
```suggestion
	pfNestingModeUnknown pfNestingMode = iota
	pfNestingModeList    pfNestingMode 
	pfNestingModeSet     pfNestingMode 
	pfNestingModeSingle  pfNestingMode 
```

Reply:
We do match the TF type here. I've added a comment to clarify that.